examples/mem-alloc-test/src/main.rs

```diff
@@ -0,0 +1,275 @@
+        ptr = ptr + scan_interval;
+    }
+}
+
+fn worker_thread(tid: i32, shared_mutex_clone: Arc<Mutex<Counters>>,
+                 memsize: MemSize, pair_clone: Arc<(Mutex<bool>, Condvar)>) {
+
+    /* Wait for all the threads to be created and then start together */
+    wait_per_thread(pair_clone);
+
+    let mut count = 0;
+    let mut tot_time_ns = 0;
+
+    /* Once the thread's allocation and deallocation operations begin, we
+     * shouldn't take any lock as the allocator that we trying to test is a
+     * multithreaded allocator and we should allow as many threads as possible
+     * to get the lock.
+     */
+    loop {
+        /* Create a random size depending on the memory type */
+        let (scan_interval, size, limit) = match memsize {
+            MemSize::Large => {
+                /* buffer size will be from 1GB to 4GB */
+                (SCAN_INTERVAL_LARGE_THREAD, TO_GB * get_random_num(LARGE_THREAD_MEM_START ,
+                                                   LARGE_THREAD_MEM_END), LIMIT_LARGE_THREAD)
+            }
+            MemSize::Medium => {
+                /* buffer size will be from 8MB to 128 */
+                (SCAN_INTERVAL_MEDIUM_THREAD, TO_MB * get_random_num(MEDIUM_THREAD_MEM_START ,
+                                                   MEDIUM_THREAD_MEM_END), LIMIT_MEDIUM_THREAD)
+            }
+            MemSize::Small => {
+                /* buffer size will be from 1KB to 512KB */
+                (SCAN_INTERVAL_SMALL_THREAD, TO_KB * get_random_num(
+                                                  SMALL_THREAD_MEM_START ,
+                                                  SMALL_THREAD_MEM_END), LIMIT_SMALL_THREAD)
+            }
+        };
+
+        let start_time = Instant::now();
+
+        /* Create an array of x GB where x is a random number between 1 to 4 */
+        let mut large_vector = Vec::with_capacity(size);
+        large_vector.resize(size, 0);
+
+        /* Traverse and access the entire buffer so that pages are allocated */
+        traverse_buffer(&mut large_vector, scan_interval);
+
+        /* deallocate */
+        drop(large_vector);
+
+        /* calculate the metrics */
+        let end_time = Instant::now();
+        let duration = end_time.duration_since(start_time);
+        tot_time_ns += duration.as_nanos();
+
+        count = count + 1;
+        if (count >= limit) {
+            break;
+        }
+    }
+
+    /* At this point the thread's allocation and deallocation operations are
+     * completed and hence it is okay to take a lock.
+     */
+
+    let avg_thread_latency = tot_time_ns as f64 / count as f64;
+
+    let shared_mutex_clone_2 = Arc::clone(&shared_mutex_clone);
+    calculate_and_print_stat(shared_mutex_clone_2, tid, &memsize, avg_thread_latency);
+
+
+}
+
+fn spawn_threads(thread_count: i32) {
+    let mut handles = vec![];
+    let shared_variable = Arc::new(Mutex::new(Counters{
+            avg_duration_large_thread:0.0,
```

Comment:
Minor: I think the preferred formatting for these is like:
```
avg_duration_large_thread: 0.0,
```
(with no space before the `:` but one space after the `:`. I think this is one of things that `cargo fmt` will format for you.